pkg/loop/internal/broker.go

```diff
@@ -66,8 +65,8 @@ type brokerExt struct {
 	BrokerConfig
 }
 
-// named returns a new [*brokerExt] with name added to the logger.
-func (b *brokerExt) named(name string) *brokerExt {
+// withNamedLogger returns a new [*brokerExt] with name added to the logger.
+func (b *brokerExt) withNamedLogger(name string) *brokerExt {
```

Comment:
Was this confusing before? b/c the fact that only the logger is being named is sort of an implementation detail and left out by design. Would `withName` be more clear?
```suggestion
func (b *brokerExt) withName(name string) *brokerExt {
```

Reply:
The previous name made me think we were setting a property `name` of broker ext directly, which seemed sensible and I moved on. I ran into the method later on and was surprised to see it do all this:
```
	bn := *b
	bn.Logger = logger.Named(b.Logger, name)
	return &bn
```

I'd  like the method name to reflect what it's actually doing, which is creating a named logger and setting the new brokerExt with it.

Reply:
> I'd like the method name to reflect what it's actually doing, which is creating a named logger and setting the new brokerExt with it.

But from the caller's perspective it is not necessarily doing that. It is returning a named `brokerExt`, and the implementation details are not important and may change.